src/PubSubClient.cpp

```diff
@@ -196,7 +196,7 @@ bool PubSubClient::connect(const char* id, const char* user, const char* pass, c
             while (!_client->available()) {
                 yield();
                 unsigned long t = millis();
-                if (t - lastInActivity >= static_cast<unsigned long>(this->socketTimeout * 1000UL)) {
+                if (t - lastInActivity >= this->socketTimeout * 1000UL) {
```

Comment:
I am not 100% sure about the precedence of operators here.
So maybe best to use (), just to be sure?

```c++
if ((t - lastInActivity) >= (this->socketTimeout * 1000UL))
```

Reply:
I always find it hard to check these kind of time diff tests as they rely on being equal unsigned types and how they overflow.

In my own project, I have made these years ago, so I don't need to think about it:
https://github.com/letscontrolit/ESPEasy/blob/0184366965210b26a283f1297a08d0d25be44865/src/src/Helpers/ESPEasy_time_calc.h#L23-L52

You could also add those, but please use slightly different names or make them static in this class if you keep the names the same. (or else I will run into compile issues)

Reply:
> I am not 100% sure about the precedence of operators here. So maybe best to use (), just to be sure?
> 
> ```c++
> if ((t - lastInActivity) >= (this->socketTimeout * 1000UL))
> ```

100% sure, not needed. Otherwise we would have lots of problems at other locations ...
`socketTimeout` is `uint16_t` but because we multiply by an `UL` it get's converted to `uint32_t` and has the same size like `t`(`unsigned long`= `uint32_t` in 32-Bit systems).